Sources/Clappr_iOS/Classes/Base/Layers/BackgroundLayer.swift

```diff
@@ -0,0 +1,25 @@
+import UIKit
+
+final class BackgroundLayer: UIView {
+    
+    init() {
+        super.init(frame: CGRect.zero)
+        backgroundColor = UIColor.black
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func attach(to view: UIView) {
+        view.addSubview(self)
+        
+        translatesAutoresizingMaskIntoConstraints = false
+        
+        widthAnchor.constraint(equalTo: view.widthAnchor).isActive = true
+        heightAnchor.constraint(equalTo: view.heightAnchor).isActive = true
+        
+        centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
```

Comment:
Maybe we can replace this code with the `UIView+Ext`, with the `anchorInCenter` and `setWidthAndHeight` methods. What do you think?

Reply:
In this case, `anchorInCenter` works great. But `setWidthAndHeight` would make the layer size constant, but what we actually need here is for it to match its container (the Core view) size. So we kept the constraints definition together instead of using only one extension method, as we thought it makes the code more readable.

We also agreed that this extension needs a little bit of cleaning up. It's kinda messy right now with methods with multiple responsibilities and using the visual constraint language, just to name a few issues with it. The initial idea was to create a new PR with these improvements and then update this code.

Reply:
Nice. I agree with you
=]